tests/assignment_tests.rb

```diff
@@ -0,0 +1,32 @@
+class AssignmentTest < MiniTest::Test
+
+  def setup
+    @course ||= Course.create(name: rand_a_z, course_code: rand_course_code)
+  end
+
+  def test_assignments_have_course_id_and_name_and_percent_of_grade
+    assignment1 = Assignment.find_or_create_by!(course_id: 'ZZZ999', name: 'Destroy C-137', percent_of_grade: '89', due_at: '2017-01-01 21:06:59.001', active_at: '2016-10-18 21:06:59.001')
+    assignment2 = Assignment.create(due_at: '2014-01-18 21:06:59.001', active_at: '2013-01-18 21:06:59.001')
+
+    assert assignment1.persisted?
+    assert assignment2.errors.full_messages.include?("Course can't be blank")
+    assert assignment2.errors.full_messages.include?("Name can't be blank")
+    assert assignment2.errors.full_messages.include?("Percent of grade can't be blank")
+  end
+
+  def test_assignment_name_is_unique_within_given_course_id
+    course1 = Course.create(name: 'course', course_code: rand_course_code)
+    course2 = Course.create(name: 'Sharp Pointy Things & Other Reasons to Become a Gelatinous Cube', course_code: rand_course_code)
+
+    assignment1 = Assignment.find_or_create_by!(course_id: 'ZZZ000', name: 'Aztec Tomb', percent_of_grade: '76', due_at: '2018-01-18 21:06:59.001', active_at: '2017-08-18 21:06:59.001') #course1.id might be suff
```

Comment:
This one, ideally, shouldn't have saved since `course_id` (one thinks) is a foreign key and should reference a primary key over on course.

If your validation just verifies the presence of the id, this will save, but it doesn't really work. You could validate the presence of `:course` itself to verify that there is not only something in that field, but that it points to a proper course.